apps/devhub/tests/test_views_ownership.py

```diff
@@ -92,11 +92,9 @@ def test_no_license(self):
 
     def test_no_license_required_for_unlisted(self):
         self.addon.update(is_listed=False)
-        self.addon.latest_version.update(version=None)
         data = self.formset(builtin='')
         response = self.client.post(self.url, data)
         assert response.status_code == 302
-        assert not self.addon.latest_version.version
```

Comment:
This change isn't linked to the bug itself, but was done because:
1/ adding strict mode for mysql is needed for the `TestThemeForm.test_long_author_or_display_username` to properly fail if the database fields are too small (if it's not in strict mode, it just truncate the data without raising a DataError like shown in the bug)
2/ testing the `latest_version` has nothing to do with the test, not sure why I wrote that in the first place (maybe some copy paste when I did it in https://github.com/magopian/olympia/commit/125a4b6c7427fd6e84d45e639e9963205b78cd3e#diff-e6c4b37ec2686327fda363a0d8a6dde7R95 ?)